### PR TITLE
Editing Email Template

### PR DESCRIPTION
**Pull request by jewang25**
### Summary 
- added basic functionality for editing email template in the email modal. 
- can click edit button to change the html for the template and then save the new template

This pull request is the first step towards implementing feature to edit emails on email modal

- implemented the basic structure of this feature by adding a textfield and saving the new template
- work in progress: still need to receive designs for this feature and implement the new design 

### Test Plan 

- testing in progress: tested by printing out the template once the email sent button is clicked and seeing if the changes were saved
- testing both match and no match templates
- testing if the new template is preserved when back button is clicked

### CODE REVIEW

**Review by rgu0114 — COMMENTED** [review]
This looks rly good Jessica, nice work! I like how you used an inlined function to replace the html pretty cleanly, and I don't see any issues with this PR. Could you also attach screenshots/recordings of what this looks like?

File: frontend/src/modules/EditZing/Components/EmailModal.tsx [hunk 1]
```diff
@@ -151,6 +152,36 @@ export const EmailModal = ({
...
+        <Button
+          onClick={() => {
+            setStep(4)
+          }}
+          color="secondary"
+          variant="outlined"
+        >
+          Edit
+        </Button>
+      )
```
Comment (mluo24) [inline frontend/src/modules/EditZing/Components/EmailModal.tsx]: Change this part to the designs

File: frontend/src/modules/EditZing/Components/EmailEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    <Box>
+      <Box sx={TitleSx}>Email Edit</Box>
+      <TextField
+        multiline={true}
+        defaultValue={replacedHtml}
+        onChange={(event) => (replacedHtml = event.target.value)}
+      />
+      <SaveButton />
+    </Box>
+  )
```
Comment (mluo24) [inline frontend/src/modules/EditZing/Components/EmailEdit.tsx]: I'm not sure if it's the fault of this component or somewhere else, either way the styling is a little messed up.

File: frontend/src/modules/EditZing/Components/EmailEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const copyEmailTemplate = () => {
+    let copied = Object.assign({}, template)
+    copied.html = replacedHtml
+    setSelectedTemplate(copied)
+    console.log('copied', copied)
+    console.log(template)
+  }
+
+  const handleClick = () => {
+    console.log('clicked')
```
Comment (mluo24) [inline frontend/src/modules/EditZing/Components/EmailEdit.tsx]: You can delete the console.logs

File: frontend/src/modules/EditZing/Components/EmailEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        onClick={() => {
+          handleClick()
+        }}
+        color="secondary"
+        variant="outlined"
+      >
+        Save
+      </Button>
+    )
+  }
```
Comment (mluo24) [inline frontend/src/modules/EditZing/Components/EmailEdit.tsx]: This doesn't give any feedback when you click. I didn't realize anything got changed until I pressed back and the changes were there.

**Review by mluo24 — CHANGES_REQUESTED** [review]
Hi Jessica, thanks for taking on this task. I've mentioned before that there are now designs that you can reference here: [https://www.figma.com/file/Dc6q3OOEV4JnHLs5T7VRXZ/Prachi's-Working-File?node-id=1%3A38561&t=unUrNvsIxW1cVBB5-0](https://www.figma.com/file/Dc6q3OOEV4JnHLs5T7VRXZ/Prachi%27s-Working-File?node-id=1%3A38561&t=unUrNvsIxW1cVBB5-0)

There are also some visual bugs, like this one: 
image

Other than that, the functionality seems alright, besides a small nit I made about how the saving does not really show feedback that you have actually saved something. Good work on that!

**Review by mluo24 — APPROVED** [review]
Thanks for fixing the styling. I tested this out, and everything looks good! I think one noteworthy part is that the subject can't be edited, but since this feature is functional and it's possible that that's not necessary, I think we can overlook that until they say they need it. 

Just fix lint errors and you can merge :)

**Pull request merged**